lightningd/bitcoind.c

```diff
@@ -814,12 +814,58 @@ static void fatal_bitcoind_failure(struct bitcoind *bitcoind, const char *error_
 	exit(1);
 }
 
+/* This function is used to check "chain" field from
+ * bitcoin-cli "getblockchaininfo" API */
+static char* check_blockchain_from_bitcoincli(const tal_t *ctx,
+				struct bitcoind *bitcoind,
+				char* output, const char **cmd)
+{
+	size_t output_bytes;
+	const jsmntok_t *tokens, *valuetok;
+	char *blockchain;
+	bool valid;
+
+	if (!output)
+		return tal_fmt(ctx, "Reading from %s failed: %s",
+			       bcli_args_direct(tmpctx, cmd), strerror(errno));
+
+	output_bytes = tal_count(output);
+
+	tokens = json_parse_input(cmd, output, output_bytes,
+			          &valid);
+
+	if (!tokens)
+		return tal_fmt(ctx, "%s: %s response",
+			       bcli_args_direct(tmpctx, cmd),
+			       valid ? "partial" : "invalid");
+
+	if (tokens[0].type != JSMN_OBJECT)
+		return tal_fmt(ctx, "%s: gave non-object (%.*s)?",
+			       bcli_args_direct(tmpctx, cmd),
+			       (int)output_bytes, output);
+
+	valuetok = json_get_member(output, tokens, "chain");
+	if (!valuetok)
+		return tal_fmt(ctx, "%s: had no chain member (%.*s)?",
+			       bcli_args_direct(tmpctx, cmd),
+			       (int)output_bytes, output);
+
+	blockchain = tal_strdup(tmpctx, bitcoind->chainparams->bip70_name);
```

Comment:
This `tal_strdup` is not needed.  We could do:

    if (!json_tok_streq(output, valuetok, bitcoind->chainparams->bip70_name))

Reply:
Thank you, I corrected.